mex/common/models/extracted_data.py

```diff
@@ -71,127 +61,49 @@ class ExtractedData(BaseEntity):
-                        f"got {len(had_primary_source)}"
-                    )
-            else:
-                had_primary_source = MergedPrimarySourceIdentifier(had_primary_source)
-        else:
-            raise ValueError("Missing value for `hadPrimarySource`.")
-
-        provider = get_provider()
-        identity = provider.assign(had_primary_source, identifier_in_primary_source)
-
-        # In case an identity was already found and it differs from the identifier
-        # provided to the constructor, we raise an error because it should not be
-        # allowed to change the identifier of an existing item.
-        if identifier := values.get("identifier"):
-            if isinstance(identifier, list):
-                if len(identifier) == 1:
-                    identifier = identifier[0]
-                else:
-                    raise ValueError(
-                        f"Expected one value for Identifier, got {len(identifier)}"
-                    )
-            if identity.identifier != str(identifier):
-                raise ValueError("Identifier cannot be set manually to new value.")
-
-        # In case an identity was found, we allow assigning a new stable target ID
-        # for the purpose of merging two items, except for the MEx
-        # primary source itself.
-        if stable_target_id := values.get("stableTargetId"):
-            if isinstance(stable_target_id, list):
-                if len(stable_target_id) == 1:
-                    stable_target_id = stable_target_id[0]
-                else:
-                    raise ValueError(
-                        f"Expected one value for stableTargetId, "
-                        f"got {len(stable_target_id)}"
-                    )
-            if (
-                identity.stableTargetId != str(stable_target_id)
-                and stable_target_id != MEX_PRIMARY_SOURCE_STABLE_TARGET_ID
-            ):
-                raise ValueError(
-                    "Cannot change `stableTargetId` of MEx primary source."
-                )
-
-        # update instance values
-        values["identifier"] = identity.identifier
-        values["stableTargetId"] = identity.stableTargetId
-        return values
+    if TYPE_CHECKING:  # pragma: no cover
+        # avoiding mypy issues here by typing the computed fields as property stubs
+        # https://docs.pydantic.dev/latest/api/fields/#pydantic.fields.computed_field
+
+        @property
+        def identifier(self) -> ExtractedIdentifierT:  # noqa: D102
+            ...
+
+        @identifier.setter
+        def identifier(self, value: ExtractedIdentifierT) -> None: ...
+
+        @property
+        def stableTargetId(self) -> MergedIdentifierT:  # noqa: D102, N802
+            ...
+
+        @stableTargetId.setter
+        def stableTargetId(self, value: MergedIdentifierT) -> None:  # noqa: N802
+            ...
+
+    else:
+
+        @computed_field
```

Comment:
Why not as follows, as recommended by [pydantic docs](https://docs.pydantic.dev/latest/api/fields/#pydantic.fields.computed_field) and get rid of the `TYPE_CHECKING` distinction?
```suggestion
        @property
        @computed_field
```